spec/trick_bag/io/gitignore_spec.rb

```diff
@@ -7,70 +7,45 @@
 GI=TrickBag::Io::Gitignore
 
 describe TrickBag::Io::Gitignore do
-
-  # Pass this a block to perform in the test directory.
-  def create_and_populate_test_directory(ignore_spec)
-    Dir.mktmpdir do |tmpdir|
-      Dir.chdir(tmpdir) do
-        %w(a_dir  .hidden_dir  x/y/z).each { |dirspec| FileUtils.mkdir_p(dirspec) }
-        %w(root  a_dir/sub  .hidden_dir/.hidden_file  x/y/z/deep).each { |f| FileUtils.touch(f) }
-        File.write('.gitignore', ignore_spec.join("\n"))
-        yield
-      end
+  shared_examples_for 'testing a pair' do |ignore_spec, expected_result, filename|
+    let(:ignored_files)  { GI.list_ignored_files }
+    let(:in_ignore_list) { ignored_files.include?(filename) }
+    let(:context) do
+      "ignore_spec: #{ignore_spec}, ignored files: #{ignored_files}, filename: '#{filename}', expected result: #{expected_result}, got: #{in_ignore_list}"
```

Comment:
A minor thing, but I would suggest moving the ignored files to the end of the message; they are likely to be more verbose than the other data. I would move filename to the front of the message too.